### ISSUE TITLE
fix(llama): KV-reuse determinism breaks with Metal FP accumulation differences

### ISSUE THREAD

**Issue by roryford** [roryford/BaseChatKit#966]
## Summary

`test_kvReuse_determinism` fails with Qwen3-0.6B-Q4_K_M on Apple M5. With KV cache reuse active, Turn 1 and Turn 2 produce different greedy outputs despite an identical prompt prefix.

## Root cause

When KV reuse is active, Turn 1 decodes the full prompt in a single batched call (N tokens together). Turn 2 re-decodes only the last prompt token in a 1-token batch (positions 0..N-2 from cache, position N-1 fresh). The K/V values stored in the cache are mathematically identical in both cases, but Metal GPU attention kernels use different parallel reduction strategies for batched vs single-token decode. This produces subtly different FP accumulation for position N-1, which propagates to different logits at token N.

For Qwen3-0.6B, the tokens `,` and `.` are nearly tied after ` Paris`, so the tiny FP difference flips the argmax.

## Observed

```
Turn1: [" Paris", ",", " and", " the", " capital", " of", " Germany", " is"]
Turn2: [" Paris", ".", " The", " capital", " of", " the", " United", " States"]
```

## Test

`LlamaKVPersistenceTests.test_kvReuse_determinism` — fails with Qwen3-0.6B-Q4_K_M on M5.

## Notes

- The greedy sampler non-determinism (stochastic `dist` sampler with random seed at `temperature=0`) was fixed in `ce3f20b`. Verified: the divergence persists with `llama_sampler_init_greedy()`, confirming the root cause is in the logit values themselves, not sampling.
- This is not fixable purely in application code. Options: document the limitation, use a tolerance-based assertion, or investigate llama.cpp Metal backend flags for deterministic reduction.
- Reproduced on: Apple M5, Qwen3-0.6B-Q4_K_M.gguf, macOS 26.
